pkg/capacityscheduling/capacity_scheduling.go

```diff
@@ -0,0 +1,816 @@
+				}
+			}
+		}
+	}
+	// No potential victims are found, and so we don't need to evaluate the node again since its state didn't change.
+	if len(potentialVictims) == 0 {
+		return nil, 0, false
+	}
+
+	// If the new pod does not fit after removing all the lower priority pods,
+	// we are almost done and this node is not suitable for preemption. The only
+	// condition that we could check is if the "pod" is failing to schedule due to
+	// inter-pod affinity to one or more victims, but we have decided not to
+	// support this case for performance reasons. Having affinity to lower
+	// priority pods is not a recommended configuration anyway.
+	if fits, _, err := core.PodPassesFiltersOnNode(ctx, ph, state, pod, nodeInfo); !fits {
+		if err != nil {
+			klog.Warningf("Encountered error while selecting victims on node %v: %v", nodeInfo.Node().Name, err)
+		}
+
+		return nil, 0, false
+	}
+	var victims []*v1.Pod
+	numViolatingVictim := 0
+	sort.Slice(potentialVictims, func(i, j int) bool { return util.MoreImportantPod(potentialVictims[i], potentialVictims[j]) })
+	// Try to reprieve as many pods as possible. We first try to reprieve the PDB
+	// violating victims and then other non-violating ones. In both cases, we start
+	// from the highest priority victims.
+	violatingVictims, nonViolatingVictims := filterPodsWithPDBViolation(potentialVictims, pdbs)
+	reprievePod := func(p *v1.Pod) (bool, error) {
+		if err := addPod(p); err != nil {
+			return false, err
+		}
+		fits, _, _ := core.PodPassesFiltersOnNode(ctx, ph, state, pod, nodeInfo)
+		if !fits {
+			if err := removePod(p); err != nil {
+				return false, err
+			}
+			victims = append(victims, p)
+			klog.V(5).Infof("Pod %v/%v is a potential preemption victim on node %v.", p.Namespace, p.Name, nodeInfo.Node().Name)
+		}
+		return fits, nil
+	}
+	for _, p := range violatingVictims {
+		if fits, err := reprievePod(p); err != nil {
+			klog.Warningf("Failed to reprieve pod %q: %v", p.Name, err)
+			return nil, 0, false
+		} else if !fits {
+			numViolatingVictim++
+		}
+	}
+	// Now we try to reprieve non-violating victims.
+	for _, p := range nonViolatingVictims {
+		if _, err := reprievePod(p); err != nil {
+			klog.Warningf("Failed to reprieve pod %q: %v", p.Name, err)
+			return nil, 0, false
+		}
+	}
+	return victims, numViolatingVictim, true
+}
+
+func (c *CapacityScheduling) addElasticQuota(obj interface{}) {
+	eq := obj.(*v1alpha1.ElasticQuota)
+	oldElasticQuotaInfo := c.elasticQuotaInfos[eq.Namespace]
+	if oldElasticQuotaInfo != nil {
+		return
+	}
+
+	elasticQuotaInfo := newElasticQuotaInfo(eq.Namespace, eq.Spec.Min, eq.Spec.Max, nil)
+
+	c.Lock()
+	defer c.Unlock()
+	c.elasticQuotaInfos[eq.Namespace] = elasticQuotaInfo
+}
+
+func (c *CapacityScheduling) updateElasticQuota(oldObj, newObj interface{}) {
+	oldEQ := oldObj.(*v1alpha1.ElasticQuota)
+	newEQ := newObj.(*v1alpha1.ElasticQuota)
+	newElasticQuotaInfo := newElasticQuotaInfo(newEQ.Namespace, newEQ.Spec.Min, newEQ.Spec.Max, nil)
```

Comment:
nit: `newElasticQuotaInfo` is a function name, prefer to use `newEQInfo`

Reply:
Done

Reply:
It's still used in elasticquota.go#clone().